models/migrations/v143.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+func addProjectsInfo(x *xorm.Engine) error {
+
+	// Create new tables
+	type (
+		ProjectType      uint8
+		ProjectBoardType uint8
+	)
+
+	type Project struct {
+		ID          int64  `xorm:"pk autoincr"`
+		Title       string `xorm:"INDEX NOT NULL"`
+		Description string `xorm:"TEXT"`
+		RepoID      int64  `xorm:"INDEX"`
+		CreatorID   int64  `xorm:"NOT NULL"`
+		IsClosed    bool   `xorm:"INDEX"`
+
+		BoardType ProjectBoardType
+		Type      ProjectType
+
+		ClosedDateUnix timeutil.TimeStamp
+		CreatedUnix    timeutil.TimeStamp `xorm:"INDEX created"`
+		UpdatedUnix    timeutil.TimeStamp `xorm:"INDEX updated"`
+	}
+
+	if err := x.Sync2(new(Project)); err != nil {
+		return err
+	}
+
+	type Comment struct {
+		OldProjectID int64
+		ProjectID    int64
+	}
+
+	if err := x.Sync2(new(Comment)); err != nil {
+		return err
+	}
+
+	type Repository struct {
+		ID                int64
+		NumProjects       int `xorm:"NOT NULL DEFAULT 0"`
+		NumClosedProjects int `xorm:"NOT NULL DEFAULT 0"`
+	}
+
+	if err := x.Sync2(new(Repository)); err != nil {
+		return err
+	}
+
+	// ProjectIssues saves relation from issue to a project
+	type ProjectIssues struct {
```

Comment:
Why `ProjectIssues` but `ProjectBoard`? I think `ProjectIssue` is better.

Reply:
@lunny was choosen by vote: https://github.com/go-gitea/gitea/pull/8346#issuecomment-617445018

Reply:
I reviewed all the table names of gitea, only `issue_assignees` has a plural form, I think we should keep singular form for all the table names to not let people confusing. If most people vote that, we should know the reason.

Reply:
I think it should be `ProjectIssue` - I understand the desire to make these plural but I think we should just keep these singular if nothing else is plural.